allow setting belongs_to `optional: true`

vollnhals opened the pull request:
We have an audit model that gets used by all model classes. Some models are not per tenant (global). So we would like to say, that tenant is optional in audit class

```
class Audit < ApplicationRecord
  multi_tenant :tenant, optional: true
end
```

So we want to allow setting the `optional: true` on the belongs_to.

lfittl commented:
@vollnhals I wonder if it might not be better to include the `multi_tenant` annotation separately in each model then.

The `belongs_to` is one minor side effect of adding `multi_tenant` to a class. The more important change based on it happens in query rewriting, which you wouldn't want on a class thats not tenant-specific.

Maybe you can move the `multi_tenant` definition into a separate concern that gets included in those classes that are actually associated to a tenant?

vollnhals commented:
@lfittl I cannot add the `multi_tenant` annotation separately in each model, because the `Audit` model belongs to many class polymorphically and the `Audit` model needs to have the `multi_tenant` behavior.

I will try to explain in more detail.

We use the `audited` gem to audit changes to application models. The gem allows you to provide / customize the `Audit` class. But there is only this one class, we cannot provide two versions (one with multi_tenant, one without).

We added `multit_tenant :tenant` to the audit class and it works for all the application models that are also multi tenanted.

But we have some models that are not multi tenant. In these cases the corresponding audit record also cannot have a `tenant_id`. So now Rails complains with a validation error that the `belongs_to` that was automatically added, has no value present.

The only solution I see is to disable that validation.

So that's why I added `optional` as an allowed argument.

Would you reconsider adding that change or do you see another better way to do it?

serprex commented:
Closing in favor of #147 which includes tests
